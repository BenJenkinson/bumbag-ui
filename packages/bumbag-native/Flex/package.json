{
  "name": "bumbag-native/Flex",
  "private": true,
  "sideEffects": false,
  "main": "../lib/Flex",
  "module": "../es/Flex",
  "types": "../ts/Flex"
}