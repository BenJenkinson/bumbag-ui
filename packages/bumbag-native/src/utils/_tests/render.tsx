import * as React from 'react';
import { ThemeConfig } from '../../types/theme';
import { render } from '@testing-library/react';
// @ts-ignore
import { Provider } from '../../Provider';

export default (
  Component: any,
  { colorMode = 'default', theme = {} }: { colorMode?: string; theme?: ThemeConfig } = {}
) =>
  render(
    <Provider platform="native" theme={theme} colorMode={colorMode}>
      {Component}
    </Provider>
  );
