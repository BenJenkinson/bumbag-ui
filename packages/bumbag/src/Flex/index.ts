import * as styles from './styles';

export * from './Flex';
export { styles as flexStyles };
