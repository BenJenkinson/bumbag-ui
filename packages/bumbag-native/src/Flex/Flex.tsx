import { ViewProps as RNViewProps } from 'react-native';
import { createComponent, createElement, createHook } from 'bumbag/utils';

import { Box, BoxProps } from '../Box';
import * as styles from './Flex.styles';

export type LocalFlexProps = {};
export type FlexProps = BoxProps & RNViewProps & LocalFlexProps;

const useProps = createHook<FlexProps>(
  (props) => {
    const boxProps = Box.useProps(props);
    return { ...boxProps };
  },
  { defaultProps: { display: 'flex', flexDirection: 'row' }, themeKey: 'native.Flex' }
);

export const Flex = createComponent<FlexProps>(
  (props) => {
    const flexProps = useProps(props);
    return createElement({
      children: props.children,
      component: styles.StyledFlex,
      htmlProps: flexProps,
    });
  },
  {
    attach: {
      useProps,
      displayName: 'native.Flex',
    },
    themeKey: 'native.Flex',
  }
);
