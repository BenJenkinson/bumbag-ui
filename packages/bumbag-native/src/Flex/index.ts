import * as flexStyles from './Flex.styles';

export * from './Flex';
export { flexStyles };
