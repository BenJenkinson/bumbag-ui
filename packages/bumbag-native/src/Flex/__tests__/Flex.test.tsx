import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { Flex } from '../Flex';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<Flex>Hello world</Flex>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(<Flex color="primary">Hello world</Flex>);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('composition', () => {
  describe('hook', () => {
    it('should return with Flex props', () => {
      const { result } = renderHook(() => Flex.useProps());
      expect(result.current).toMatchSnapshot();
    });
  });

  describe('render props', () => {
    it('should render correctly', () => {
      const { container } = render(<Flex>{(FlexProps) => <Flex {...FlexProps}>Hello world</Flex>}</Flex>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('theming', () => {
  it('Flex.styles.base should render correctly', () => {
    const { container } = render(<Flex>hello world</Flex>, {
      theme: { Flex: { styles: { base: { backgroundColor: 'red' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for color', () => {
    const { container } = render(<Flex>hello world</Flex>, {
      theme: { Flex: { defaultProps: { color: 'primary' } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
