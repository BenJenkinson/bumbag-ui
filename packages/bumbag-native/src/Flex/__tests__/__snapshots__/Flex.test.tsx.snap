// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composition hook should return with Flex props 1`] = `
Object {
  "colorMode": "default",
  "elementRef": [Function],
  "ref": [Function],
  "style": Array [
    Object {
      "display": "flex",
      "flexDirection": "row",
    },
    Object {},
  ],
}
`;

exports[`composition render props should render correctly 1`] = `
<div
  class="css-view-1dbjc4n"
  style="display: flex; flex-direction: row;"
>
  <div
    class="css-text-901oao"
    dir="auto"
    style="color: rgb(33, 33, 33); font-size: 16px;"
  >
    Hello world
  </div>
</div>
`;

exports[`defaultProps should render correctly for color 1`] = `
<div
  class="css-view-1dbjc4n"
  style="color: rgb(87, 79, 235); display: flex; flex-direction: row;"
>
  <div
    class="css-text-901oao"
    dir="auto"
    style="color: rgb(33, 33, 33); font-size: 16px;"
  >
    Hello world
  </div>
</div>
`;

exports[`props should render correctly 1`] = `
<div
  class="css-view-1dbjc4n"
  style="display: flex; flex-direction: row;"
>
  <div
    class="css-text-901oao"
    dir="auto"
    style="color: rgb(33, 33, 33); font-size: 16px;"
  >
    Hello world
  </div>
</div>
`;

exports[`props should render correctly with CSS props 1`] = `
<div
  class="css-view-1dbjc4n"
  style="color: rgb(87, 79, 235); display: flex; flex-direction: row;"
>
  <div
    class="css-text-901oao"
    dir="auto"
    style="color: rgb(33, 33, 33); font-size: 16px;"
  >
    Hello world
  </div>
</div>
`;

exports[`theming Flex.styles.base should render correctly 1`] = `
<div
  class="css-view-1dbjc4n"
  style="background-color: rgb(255, 0, 0); display: flex; flex-direction: row;"
>
  <div
    class="css-text-901oao"
    dir="auto"
    style="color: rgb(33, 33, 33); font-size: 16px;"
  >
    Hello world
  </div>
</div>
`;
